### PR TITLE
Rename DisjointSet and resolve errors

### PR DESCRIPTION
**Pull request by c-p-murphy**
Addresses #694 and resolves error when pushing a repeated element to a `DisjointSet`:

```julia
julia> ds = DisjointSets('a':'e')
DisjointSets{Char} with 5 elements:
  'a'
  'b'
  'c'
  'd'
  'e'

julia> ds.intmap
Dict{Char,Int64} with 5 entries:
  'a' => 1
  'c' => 3
  'd' => 4
  'e' => 5
  'b' => 2

julia> ds.internal
IntDisjointSets{Int64}([1, 2, 3, 4, 5], [0, 0, 0, 0, 0], 5)

julia> push!(ds, 'b')
'b': ASCII/Unicode U+0062 (category Ll: Letter, lowercase)

julia> ds.intmap
Dict{Char,Int64} with 5 entries:
  'a' => 1
  'c' => 3
  'd' => 4
  'e' => 5
  'b' => 6

julia> ds.internal
IntDisjointSets{Int64}([1, 2, 3, 4, 5, 6], [0, 0, 0, 0, 0, 0], 6)
```

### CODE REVIEW

**Comment by oxinabox** [comment]
Cool so we need to `@deprecate_binding DisjointSets DisjointSet`
and `@deprecate DisjointSets{T}(xs::T...) where T = DisjointSet{T}(xs)`

and test those depracations behave as expected

File: test/test_disjoint_set.jl [hunk 1]
```diff
@@ -162,4 +162,9 @@
         end
     end
 
+    @testset "Test deprecations from #700" begin
+        @test DisjointSets(1:10) isa DisjointSet
+        @test IntDisjointSets(10) isa IntDisjointSet
+        @test DisjointSets(1:10...) == DisjointSet(1:10)
```
Comment (oxinabox) [inline test/test_disjoint_set.jl:168]:
```suggestion
        @test DisjointSets(1:10...) == DisjointSet(1:10)
        @test typeof(DisjointSet(1:10)) <: DisjointSets  # check old name can still be used for type

```

**Review by oxinabox — APPROVED** [review]
LGTM.
One last suggestion on tests and then we can merge

**Comment by oxinabox** [comment]
thanks, awesome

**Comment by ChrisRackauckas** [comment]
Why rename? This seems like pointless code churn.

**Pull request merged**